### ISSUE TITLE
Disallow CHPL_TARGET_COMPILER=clang-included when CHPL_LLVM=system

### ISSUE THREAD

**Issue by BryantLam** [chapel-lang/chapel#17598]
Related #17584 and [this post](https://chapel.discourse.group/t/4554/5).

As of 1.24.1,
```
CHPL_LLVM=system
CHPL_TARGET_COMPILER=clang-included`
```
behaves like (identically to?)
```
CHPL_LLVM=system
CHPL_TARGET_COMPILER=clang
```

If `CHPL_LLVM=system`, I think `CHPL_TARGET_COMPILER=clang-included` should be an illegal value.

**Comment by xgupta** [comment]
I will work on this issue.

**Comment by bradcray** [comment]
@mppf: I think that you intended for this to be closed with #17800, but that either GitHub doesn't permit chaining of auto-resolves, or the syntax wasn't quite right, is that correct?

**Comment by mppf** [comment]
Yes, I meant to close this with #17800, thanks for pointing it out.
